### PR TITLE
Dump docker image list

### LINKED ISSUE #88011: Dump list of test containers on Windows test nodes
When a test run completes it would be useful to have a list of all the test containers that were pulled onto the node - see https://github.com/kubernetes-sigs/windows-testing/pull/126#discussion_r377413135 for the motivation. Let's add a call to `docker image list` at https://github.com/kubernetes/kubernetes/blob/master/cluster/log-dump/log-dump.sh#L249 - just after `export-windows-docker-event-log` seems like it should work fine. We can dump the output to a file and gather it from each node just like we do for docker.log.

### PR DESCRIPTION
**Pull request by YangLu1031**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
Dump docker image list after e2e test job to provide container images info for prepulling.

**Which issue(s) this PR fixes**:

Fixes #88011

**Does this PR introduce a user-facing change?:**
```release-note
NONE
```

### CODE REVIEW

**Comment by pjh** [comment]
Please add a NONE release note: https://github.com/kubernetes/community/blob/master/contributors/guide/release-notes.md#applying-a-release-note

**Comment by YangLu1031** [comment]
@pjh I can run the gcloud compute ssh xxx cmd against windows nodes in Asia zones, so verified it works well.

**Comment by pjh** [comment]
@eparis @mikedanese could you PTAL and approve when you have a chance? This will help us investigate some flaky test runs. Thanks.

**Comment by YangLu1031** [comment]
@spiffxp @zmerlynn Could you help to take a look? Thanks!

File: cluster/log-dump/log-dump.sh [hunk 1]
```diff
@@ -218,6 +218,25 @@ function export-windows-docker-event-log() {
...
+function export-windows-docker-images-list() {
+    local -r node="${1}"
+
+    local -r powershell_cmd="powershell.exe -Command \"\$logs=\$(docker image list); \$logs | Out-File -FilePath '${WINDOWS_LOGS_DIR}\\docker_images.log'\""
+
+    # Retry up to 3 times to allow ssh keys to be properly propagated and
+    # stored.
+    for retry in {1..3}; do
+      if gcloud compute ssh --project "${PROJECT}" --zone "${ZONE}" "${node}" \
+        --command "$powershell_cmd"; then
```
Comment (mikedanese) [inline cluster/log-dump/log-dump.sh]:
```suggestion
        --command "${powershell_cmd}"; then
```
Use braced variable expansions.

https://google.github.io/styleguide/shellguide.html#s5.6-variable-expansion

**Pull request merged**